jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/ServletConstants.java

```diff
@@ -1,53 +1,65 @@
+    public static final String SERVLET_CONTEXT_ATTRIBUTE_LISTENER = "ServletContextAttributeListener";
+    public static final String SERVLET_CONTEXT_ATTRIBUTE_LISTENER_FQ_NAME = "jakarta.servlet.ServletContextAttributeListener";
+    public static final String SERVLET_REQUEST_LISTENER = "ServletRequestListener";
+    public static final String SERVLET_REQUEST_LISTENER_FQ_NAME = "jakarta.servlet.ServletRequestListener";
+    public static final String SERVLET_REQUEST_ATTRIBUTE_LISTENER = "ServletRequestAttributeListener";
+    public static final String SERVLET_REQUEST_ATTRIBUTE_LISTENER_FQ_NAME = "jakarta.servlet.ServletRequestAttributeListener";
+    public static final String HTTP_SESSION_LISTENER = "HttpSessionListener";
+    public static final String HTTP_SESSION_LISTENER_FQ_NAME = "jakarta.servlet.http.HttpSessionListener";
+    public static final String HTTP_SESSION_ATTRIBUTE_LISTENER = "HttpSessionAttributeListener";
+    public static final String HTTP_SESSION_ATTRIBUTE_LISTENER_FQ_NAME = "jakarta.servlet.http.HttpSessionAttributeListener";
+    public static final String HTTP_SESSION_ID_LISTENER = "HttpSessionIdListener";
+    public static final String HTTP_SESSION_ID_LISTENER_FQ_NAME = "jakarta.servlet.http.HttpSessionIdListener";
+
+    /* @WEBServlet */
+    public static final String WEB_SERVLET = "WebServlet";
+    public static final String WEB_SERVLET_FQ_NAME = "jakarta.servlet.annotation.WebServlet";
+    public static final String HTTP_SERVLET = "HttpServlet";
+
+    /* @WEBFilter */
+    public static final String WEBFILTER = "WebFilter";
+    public static final String WEBFILTER_FQ_NAME = "jakarta.servlet.annotation.WebFilter";
+    public static final String FILTER = "Filter";
+    public static final String FILTER_FQ_NAME = "jakarta.servlet.Filter";
+
+    /* Annotation Member Value names */
+    public static final String URL_PATTERNS = "urlPatterns";
+    public static final String VALUE = "value";
+    public static final String SERVLET_NAMES = "servletNames";
+
+    /* Diagnostics fields constants */
+    public static final String DIAGNOSTIC_SOURCE = "jakarta-servlet";
+    public static final String DIAGNOSTIC_CODE = "ExtendHttpServlet";
+    public static final String DIAGNOSTIC_CODE_MISSING_ATTRIBUTE = "CompleteHttpServletAttributes";
+    public static final String DIAGNOSTIC_CODE_DUPLICATE_ATTRIBUTES = "InvalidHttpServletAttribute";
+    public static final String DIAGNOSTIC_CODE_FILTER = "ImplementFilter";
+    public static final String DIAGNOSTIC_CODE_FILTER_MISSING_ATTRIBUTE = "CompleteWebFilterAttributes";
+    public static final String DIAGNOSTIC_CODE_FILTER_DUPLICATE_ATTRIBUTES = "InvalidWebFilterAttribute";
+    public static final String DIAGNOSTIC_CODE_LISTENER = "ImplementListener";
+    public static final DiagnosticSeverity SEVERITY = DiagnosticSeverity.Error;
+    public static final DiagnosticSeverity WARNING = DiagnosticSeverity.Warning;

```

Comment:
I wonder if we should move these severity constants to `AbstractDiagnosticsCollector`. Do we even need to set these constants ourselves?

Reply:
We can just use the one from DiagnosticsSeverity. There are multiple Constants classes defined pretty similar variables for different collectors, not sure why.

Reply:
Agreed, then lets remove this constant

Reply:
You want to remove those variables from this PR?

Reply:
Yes to use the ones from DiagnosticSeverity since we are cleaning this code up in this PR anyway